client/rest-high-level/build.gradle

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+apply plugin: 'elasticsearch.build'
+apply plugin: 'elasticsearch.rest-test'
```

Comment:
I think the trouble is https://github.com/elastic/elasticsearch/blob/master/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/StandaloneTestBasePlugin.groovy#L46-L49

I *think* if we want to repurpose RestTestPlugin so it can be applied to projects with a main class we should do something like:
```
- project.pluginManager.apply(StandaloneTestBasePlugin)
+ if (false == project.pluginManager.hasPlugin('elasticsearch.build')) {
+   project.pluginManager.apply(StandaloneTestBasePlugin)
+ }
```

Or maybe just:
```
- project.pluginManager.apply(StandaloneTestBasePlugin)
+ if (false == project.pluginManager.hasPlugin('elasticsearch.build') && false == project.pluginManager.hasPlugin('elasticsearch.standalone-test')) {
+   throw new InvalidUserDataException('elasticsearch.rest-test requires either elasticsearch.build or elasticsearch.standalone-test')
+ }
```

and then every build that needs rest tests has to ha